src/catalog/catalog.cpp

```diff
@@ -59,18 +62,29 @@ Catalog::Catalog(const std::string& directory, VirtualFileSystem* vfs) {
     registerBuiltInFunctions();
 }
 
-bool Catalog::containsTable(const Transaction* transaction, const std::string& tableName) const {
-    return tables->containsEntry(transaction, tableName);
+bool Catalog::containsTable(const Transaction* transaction, const std::string& tableName,
+    bool useInternal) const {
+    bool contains = tables->containsEntry(transaction, tableName);
+    if (!contains && useInternal) {
+        contains = internalTables->containsEntry(transaction, tableName);
```

Comment:
I thought the logic is search either Table or InternalTable.

Here you are checking Table then InternalTable. The OR semantic seems sufficient already?

If you decide to make the change. Propagate this across the PR.

Reply:
In create/query FTS, I have to query both internal/user tables.